File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/mining/smartcontract/SCExecutionProcessFactoryImpl.java
package org.palladiosimulator.blockchainsystems.core.mining.smartcontract;

import java.util.random.RandomGenerator;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.SCExecutionProcess;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.SCExecutionProcessFactory;

public class SCExecutionProcessFactoryImpl implements SCExecutionProcessFactory {

    @Override
    public SCExecutionProcess createSCExecutionProcess(String nodeId, boolean isFullNode) {
        // TODO: Adjust parameters for smart contracts
        return new SCExecutionProcessImpl(isFullNode, 5_000_000, RandomGenerator.of("Random"));
    }

}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/mining/smartcontract/SCExecutionProcessFactoryImpl.kt
package org.palladiosimulator.blockchainsystems.core.mining.smartcontract

import org.palladiosimulator.blockchainsystems.core.system.abstractions.SCExecutionProcess
import org.palladiosimulator.blockchainsystems.core.system.abstractions.SCExecutionProcessFactory
import java.util.random.RandomGenerator

/**
 * @author Davis Riedel
 */
class SCExecutionProcessFactoryImpl: SCExecutionProcessFactory {
  override fun createSCExecutionProcess(
    nodeId: String,
    isFullNode: Boolean
  ): SCExecutionProcess {
    return SCExecutionProcessImpl(isFullNode, 5000000.0, RandomGenerator.of("Random"))
  }
}